Feature/tutorial mutations

tnagorra opened the pull request:
Depends on
- https://github.com/mapswipe/mapswipe-backend/pull/23

## Changes

- Add models, queries (inputs), mutations (serializers) for tutorial
- BREAKING: Reset all migrations
- BREAKING: Remove is_draft from Project
- BREAKING: Add mandatory client_id on user resources: Project, Tutorial, ProjectAssets, etc.

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] n+1 queries
- [x] flake8 issues
- [x] `print`
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [x] tests

tnagorra committed:
Define models, queries and mutations for tutorial

tnagorra committed:
Update graphql types for project related models

- Refactor graphql common functions

tnagorra committed:
Add context on tutorial serializers

- Add test for mutation

tnagorra committed:
Make CudInput generic

tnagorra committed:
Update mutation for tutorial update

- Add update overrides for tutorial related serializers
- Add test for tutorial update
- Update parse_input_data to support overrides

tnagorra committed:
Use newer syntax for generics

tnagorra committed:
Add project_type_specifics for tutorial query and mutation

tnagorra committed:
Remove is_draft from project

tnagorra committed:
Remove all migrations

tnagorra committed:
Add client_id in UserResource

- Update serializers, gql inputs and gql types

tnagorra committed:
Add client_id on user resource

- Add client_id in serializer, gql type, gql input
- Add client_id in factories
- Remove default value for client_id
- Add validations for client_id in model and serializer
- Add option to set tutorial on a project
- Update tests

tnagorra committed:
Make migrations from scratch

tnagorra committed:
Update graphql schema

tnagorra committed:
Setup python on github actions

frozenhelium committed:
fix(tile_server): update quadKey with quad_key

frozenhelium committed:
fix(enums): add tutorial icon enum to the list

frozenhelium committed:
fix(ProjectAssetType): add file field

tnagorra committed:
Fix issue with gis widget on admin panel

tnagorra committed:
Handle None values on validate functions

tnagorra committed:
Rename scenario_id to scenario_page_number

tnagorra committed:
Update foreign key filters for project and assets

susilnem commented on apps/project/migrations/0003_alter_organization_client_id_alter_project_client_id_and_more.py:
```diff
@@ -0,0 +1,29 @@
+        ('project', '0002_initial'),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='organization',
+            name='client_id',
+            field=models.CharField(max_length=26, unique=True, validators=[apps.common.models.validate_ulid]),
+        ),
+        migrations.AlterField(
+            model_name='project',
+            name='client_id',
+            field=models.CharField(max_length=26, unique=True, validators=[apps.common.models.validate_ulid]),
+        ),
+        migrations.AlterField(
+            model_name='projectasset',
+            name='client_id',
+            field=models.CharField(max_length=26, unique=True, validators=[apps.common.models.validate_ulid]),
+        ),
+    ]
```
Do we need this migration here?
Instead, we can remove the editable=false in the initial migrations file, as we are already changing the initial migrations?
Is there any other reason creating these migrations?

susilnem commented on apps/tutorial/factories.py:
```diff
@@ -0,0 +1,53 @@
+        model = Tutorial
+
+    client_id = factory.LazyFunction(lambda: str(ULID()))
+    # name = factory.Sequence(lambda n: f"Tutorial {n}")
```
remove this?

susilnem commented on apps/tutorial/project_types/base/tutorial.py:
```diff
@@ -0,0 +1,9 @@
+import logging
+from abc import ABC
+
+from pydantic import BaseModel
+
+logger = logging.getLogger(__name__)
+
+
+class BaseTutorialTaskProperty(BaseModel, ABC): ...
```
```suggestion
from abc import ABC

from pydantic import BaseModel


class BaseTutorialTaskProperty(BaseModel, ABC): ...
```

susilnem commented on apps/tutorial/project_types/tile_map_service/base/tutorial.py:
```diff
@@ -0,0 +1,11 @@
+
+from apps.tutorial.project_types.base import tutorial as base_tutorial
+
+logger = logging.getLogger(__name__)
```
```suggestion

```

susilnem commented on apps/tutorial/serializers.py:
```diff
@@ -0,0 +1,395 @@
+        )
+
+    @typing.override
+    def create(self, validated_data: dict[typing.Any, typing.Any]):
```
Should we consider making this change atomic?

thenav56 committed:
chore(deps): upgrade sentry-sdk to 2.22.0

thenav56 committed:
fix(celery): fix crontab value issue

thenav56 committed:
feat(pre-commit): add gitleaks

thenav56 committed:
feat(sentry): add sentry config override for celery beat tasks

thenav56 committed:
fix(celery-beats): fix and add django check for beats task validation

thenav56 committed:
fix(sentry): fix site tag validation error

frozenhelium committed:
feat(docker-compose): add option to override env file

thenav56 committed:
feat(health-check): add ensure_csrf_cookie with health-check

allow csrf cookie set and backend health check from frontend

tnagorra committed:
Add status on Tutorial

- Remove is_draft from Tutorials
- Add rest-media-temp/ to dockerignore
- Expose enums on query

thenav56 committed:
fix(graphql): add custom file type to fix url issue

thenav56 committed:
feat(health-check): add custom metadata in json response

tnagorra committed:
Add name in tutorial serializer

thenav56 committed:
fix(heath-check): remove 'files' field from GitHub API response

frozenhelium committed:
Add create mutation for organization

frozenhelium committed:
Add Validate project type
